src/compas_fab/robots/constraints.py

```diff
@@ -12,32 +12,64 @@
 class BoundingVolume(object):
     """A container for describing a bounding volume.
 
+    Parameters
+    ----------
+    volume_type
+        The type of bounding volume, one of :attr:`BoundingVolume.VOLUME_TYPES`.
+    volume : :class:`compas.datastructures.Mesh` or :class:`compas.geometry.Primitive`
+        The volume can be either a :class:`compas.geometry.Box`, a
+        :class:`compas.geometry.Sphere`, or a
+        :class:`compas.datastructures.Mesh`.
+
     Attributes
     ----------
-    type: int
-        The type of the bounding volume.
-    volume: object
-        The volume can be either a `Box`, a `Sphere`, or a `Mesh`.
+    volume_type
+        The type of bounding volume, one of :attr:`BoundingVolume.VOLUME_TYPES`.
+    volume : :class:`compas.datastructures.Mesh` or :class:`compas.geometry.Primitive`
+        The volume can be either a :class:`compas.geometry.Box`, a
+        :class:`compas.geometry.Sphere`, or a
+        :class:`compas.datastructures.Mesh`.
+
+    Class attributes
+    ----------------
+    BOX
+        Box bounding volume type.
+    SPHERE
+        Sphere bounding volume type.
+    MESH
+        Mesh bounding volume type.
+    VOLUME_TYPES
+        List of supported bounding volume types.
     """
 
+    #: Box bounding volume type
     BOX = 1
+    #: Sphere bounding volume type
     SPHERE = 2
+    #: Mesh bounding volume type
     MESH = 3
 
-    def __init__(self, type, volume):
-        if type not in [self.BOX, self.SPHERE, self.MESH]:
-            raise ValueError("Type must be %d, %d or %d"
-                             % (self.BOX, self.SPHERE, self.MESH))
-        self.type = type
+    #: List of supported volume types
+    VOLUME_TYPES = (BOX, SPHERE, MESH)
+
+    def __init__(self, volume_type, volume):
+        if volume_type not in self.VOLUME_TYPES:
+            raise ValueError("Type must be %d, %d or %d" % (self.VOLUME_TYPES))
```

Comment:
Maybe best to just print out all, otherwise, adding a new volume type would break it:
```suggestion
            raise ValueError("Type must be one of {}".format(self.VOLUME_TYPES))
```